alibi_detect/cd/chisquare.py

```diff
@@ -64,16 +64,20 @@ def __init__(
             input_shape=input_shape,
             data_type=data_type
         )
-        if categories_per_feature is None:  # infer number of possible categories for each feature from reference data
+        # construct categories from the user-specified dict
+        if isinstance(categories_per_feature, dict):
+            vals = list(categories_per_feature.values())
+            if all(isinstance(v, (int, np.int16, np.int32, np.int64)) for v in vals):
+                # categories_per_feature = Dict[int, int]
+                categories_per_feature = {f: list(np.arange(v))  # type: ignore
+                                          for f, v in categories_per_feature.items()}
+            elif not all(isinstance(v, list) for v in vals):
+                raise NotImplementedError  # categories_per_feature not Dict[int, list]
+        else:  # infer number of possible categories for each feature from reference data
```

Comment:
This branch is also run if some values are floats instead of integers, might be better/more explicit to raise an error if non-ints are present rather than silently doing inference.